classes/ObjectModel.php

```diff
@@ -292,7 +292,15 @@ protected function trans($id, array $parameters = [], $domain = null, $locale =
             }
         }
 
-        $parameters['legacy'] = 'htmlspecialchars';
+        if (!isset($parameters['_raw'])) {
+            return htmlspecialchars($this->translator->trans($id, $parameters, $domain, $locale), ENT_NOQUOTES);
+        }
+
+        @trigger_error(
+            'Use _raw parameter will be removed in version 9.0, escaping translation is not the translator responsability.',
+            E_USER_DEPRECATED
+        );
+        unset($parameters['_raw']);
 
         return $this->translator->trans($id, $parameters, $domain, $locale);
```

Comment:
The deprecation and unset should be done only if the parameter is present.
The deprecation message could be improved as well.

```suggestion
        if (isset($parameters['_raw'])) {
            @trigger_error(
                'The _raw parameter is deprecated and will be removed in the next major version.',
                E_USER_DEPRECATED
            );
            unset($parameters['_raw']);
            
            return $this->translator->trans($id, $parameters, $domain, $locale);
        }
        
        return htmlspecialchars($this->translator->trans($id, $parameters, $domain, $locale), ENT_NOQUOTES);
```